src/REPLMode.jl

```diff
@@ -816,15 +820,27 @@ function do_resolve!(ctx::Context, tokens::Vector{Token})
     API.resolve(ctx)
 end
 
-function do_activate!(tokens::Vector{Token})
+function do_activate!(env::Union{EnvCache,Nothing}, tokens::Vector{Token})
     if isempty(tokens)
         return API.activate()
     else
-        token = popfirst!(tokens)
-        if !isempty(tokens) || !(token isa String)
+        path = popfirst!(tokens)
+        if !isempty(tokens) || !(path isa String)
             cmderror("`activate` takes an optional path to the env to activate")
         end
-        return API.activate(abspath(token))
+        # `pkg> activate path` does the following
+        # 1. if path exists, activate that
+        # 2. if path exists in deps, and the dep is deved, activate that path
+        # 3. activate the non-existing directory (e.g. as in `pkg> activate .` for initing a new env)
+        if Types.isdir_windows_workaround(path)
+            API.activate(abspath(path))
+        elseif env !== nothing && haskey(env.project, "deps") && haskey(env.project["deps"], path) &&
+            haskey(env.manifest, path) && haskey(env.manifest[path][1], "path") &&
+            env.manifest[path][1]["path"] != nothing
```

Comment:
Is there a better way to make sure there is a deved path to activate than these 3 lines?

Reply:
The `1` here is not correct in case there are multiple packages with the same name in the Manifest.
Get the `uuid` from `env.project["deps"][path]` and use `info = manifest_info(env, uuid)` instead.
Then look if `info["path"]` exist.